cargo-hwtracer

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+#
+# Builds ykbf against the Yorick hardware tracing JIT.
+#
+# Ideally this wrapper script wouldn't be necessary and we'd just have the user
+# pass `--features hw-tracer` to regular Cargo and we'd set flags in a
+# `build.rs` depending upon the `YK_DIR` and `CARGO_FEATURE_hw-tracer`
+# environments.
+#
+# The problem is that (at the time of writing) `build.rs` doesn't let us pass
+# arbitrary `RUSTFLAGS` to rustc. If we pass `-Z extra-link-arg` to Cargo, we
+# can pass *linker* flags, but we still have no way to tack on `-C tracer=hw`.
+# So without this wrapper, we'd be asking the user to write commands like:
+#
+# RUSTFLAGS="-C tracer=hw" cargo -Z extra-link-arg build
+#
+# or
+#
+# RUSTFLAGS="-C tracer=hw -C link-arg '-L/some/path -Wl,-rpath=/some/path" cargo build
+#
+# With this wrapper, they just write:
+#
+# ./cargo-hwtracer build
+#
+# And then there's the issue of finding `ykrt`, which will live at a different
```

Comment:
Silly question: why aren't we pointing to `ykrt` on github rather than (AIUI) requiring users to have a local checkout?

Reply:
The dual workspace setup means that nothing quite works as it should any more.

If we pointed to github, it wouldn't run the `xtask` build system, and `libykshim.so` wouldn't be built. So we have to build it externally.

There might still be a way to make it better, but I'd like to look at in in another PR.

Reply:
Could `build.rs` download and build it? But, fair enough, let's do it in the next PR.

Reply:
We could try something like that, yeah.